### PR TITLE
[MRG] changed the API and docstring of map_threshold

### PR DESCRIPTION
**Pull request by bthirion**
Addresses #221

### CODE REVIEW

**Comment by bthirion** [comment]
Please review.

**Comment by KamalakerDadi** [comment]
I am looking at the issue. I did not understand this point "When fpr is used, it should not require an image". Could you say more details on this in this PR ?

**Comment by KamalakerDadi** [comment]
>Well, when you simply control the FPR, the threhsold does not depend on the data you entered: it only depdends on the targeted p-value. In that sense, the function should not even require an image. Does that makes sense ?

function here you mean ```map_threshold``` ?

File: nistats/thresholding.py [hunk 1]
```diff
@@ -10,21 +10,36 @@
...
+        return (z_vals_[pos][-1] - 1.e-12)
     else:
         return np.infty
 
 
-def map_threshold(stat_img, mask_img=None, threshold=.001,
+def map_threshold(stat_img, mask_img=None, level=.001,
                   height_control='fpr', cluster_threshold=0):
-    """ Threshold the provided map
+    """ Compute the required threhsold level and return the thresholded map
```
Comment (KamalakerDadi) [inline nistats/thresholding.py]: threshold

File: nistats/thresholding.py [hunk 2]
```diff
@@ -34,15 +49,21 @@ def map_threshold(stat_img, mask_img=None, threshold=.001,
     mask_img : Niimg-like object, optional,
         mask image
 
-    threshold: float, optional
-        cluster forming threshold (either a p-value or z-scale value)
+    level: float, optional
+        number controling the thresholding (either a p-value or z-scale value)
+        Not te be confused with the z-scale threshold: level can be a p-values,
```
Comment (KamalakerDadi) [inline nistats/thresholding.py]: to

File: nistats/thresholding.py [hunk 3]
```diff
@@ -10,21 +10,36 @@
 
 
 def fdr_threshold(z_vals, alpha):
-    """ return the BH fdr for the input z_vals"""
+    """ return the Benjamini-Hochberg FDR threshold for the input z_vals
+    
+    Parameters
+    ----------
+    z_vals: array,
+            a set of z-variates from which an FDR
```
Comment (KamalakerDadi) [inline nistats/thresholding.py]: I would write "a set of z-variates from which an FDR is calculated."

**Comment by KamalakerDadi** [comment]
I think this changes deserves to add in change log.

**Comment by KamalakerDadi** [comment]
Because there is no deprecation warning to remove threshold parameter

**Comment by bthirion** [comment]
Will probably merge with in the next 24hours unless I hear otherwise.

File: nistats/thresholding.py [hunk 4]
```diff
@@ -10,48 +10,98 @@
...
-    """ Threshold the provided map
+    """ Compute the required threshold level and return the thresholded map
 
     Parameters
     ----------
-    stat_img : Niimg-like object,
+    stat_img : Niimg-like object or None, optional
        statistical image (presumably in z scale)
+       whenever height_control is 'fpr' or 'none', 
+       stat_img=None os acceptable.
```
Comment (KamalakerDadi) [inline nistats/thresholding.py]: is

Comment (KamalakerDadi) [reply nistats/thresholding.py]: May be: "For height_control='fpr', stat_img has no effect and can be given as None."

File: nistats/thresholding.py [hunk 5]
```diff
@@ -10,48 +10,98 @@
...
+    """ Compute the required threshold level and return the thresholded map
 
     Parameters
     ----------
-    stat_img : Niimg-like object,
+    stat_img : Niimg-like object or None, optional
        statistical image (presumably in z scale)
+       whenever height_control is 'fpr' or 'none', 
+       stat_img=None os acceptable. 
+       If it is 'fdr' or 'bonferroni', an error is raised
```
Comment (KamalakerDadi) [inline nistats/thresholding.py]: an error is raised if stat_img is not provided.

File: nistats/thresholding.py (hunk 4 above)
Comment (bthirion) [reply nistats/thresholding.py]: It is actually a bit more subtle: when the image is provided, a thresholded version is returned. For this readons, I prefer to keep the current formulation.

**Comment by bthirion** [comment]
Thx for the review ! 
Will merge that one soon.

**Pull request merged**